packages/consensus/src/validation/coinbase.cairo

```diff
@@ -112,7 +110,13 @@ fn validate_coinbase_witness(witness: Span<ByteArray>) -> Result<(), ByteArray>
 
 /// Return BTC reward in SATS
 fn compute_block_reward(block_height: u32) -> u64 {
-    shr(5000000000_u64, (block_height / 210000_u32))
+    let mut result: u64 = 5_000_000_000;
+
+    for _ in 0..block_height / 210_000 {
```

Comment:
Afaik the conditions are not cached (need to check). Would it help to have `let num_halvings = block_height / 210_000` before?